Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift

```diff
@@ -621,6 +621,12 @@ extension BrowserViewController: WKNavigationDelegate {
                 }
             }
         }
+        
+        // Ecosia: Allow MarketplaceKit requests for iOS 17.4+
+        if #available(iOS 17.4, *), url.scheme == "marketplace-kit" {
```

Comment:
Isn't leaving here at the end risky that it will fall into a previous condition?

For example, given the scheme does not contain firefox, it would fall within `!(url.scheme?.contains("firefox") ?? true)` just above and be execute as an external link, wouldn't it?

Reply:
💯 happy and well spotted! I did just move it above as indeed we do want to basically let the OS do its own navigation